pandas/_libs/tslib.pyx

```diff
@@ -509,6 +514,8 @@ cpdef array_to_datetime(
         datetime py_dt
         tzinfo tz_out = None
         bint found_tz = False, found_naive = False
+        char inferred_format[100]
+        int inferred_format_len
```

Comment:
hi @WillAyd - I don't have much experience with C, is this an OK way to do it?

What I'm trying to do is:
- make a character array `inferred_format` in which to record the format which is guessed within `parse_iso_8601_datetime`
- record the length of the inferred format into `inferred_format_len`

and then compare `format` with `inferred_format[:inferred_format_len]`